### PR TITLE
Basic analytics

### PR DESCRIPTION
**Pull request by nwalters512**
Implements a basic analytics page for course instances, with per-student analytics. Provides a framework on which to add more data + analytics.

### CODE REVIEW

**Review by Binary-Eater — APPROVED** [review]
Looked over code and tested functionality

And ran eslint :+1:

File: src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// const ERR = require('async-stacktrace')
```
Comment (JParisFerrer) [inline src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js]: nit: remove

File: src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const router = require('express').Router({ mergeParams: true })
+const { sqlLoader } = require('@prairielearn/prairielib')
+const dbDriver = require('../../dbDriver')
+const asyncErrorHandler = require('../../asyncErrorHandler')
+// const checks = require('../../auth/checks')
```
Comment (JParisFerrer) [inline src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js]: nit: remove

File: src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const sql = sqlLoader.loadSqlEquiv(__filename)
+
+router.get(
+  '/',
+  asyncErrorHandler(async (req, res, _next) => {
+    const { courseInstanceId } = req.params
```
Comment (JParisFerrer) [inline src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js:17]: auth protect this route

File: src/pages/courseInstanceStudentAnalytics/courseInstanceStudentAnalytics.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+const dbDriver = require('../../dbDriver')
+const asyncErrorHandler = require('../../asyncErrorHandler')
+// const checks = require('../../auth/checks')
+
+const sql = sqlLoader.loadSqlEquiv(__filename)
+
+router.get(
+  '/',
+  asyncErrorHandler(async (req, res, _next) => {
+    const courseInstance = (await dbDriver.asyncQuery(
```
Comment (JParisFerrer) [inline src/pages/courseInstanceStudentAnalytics/courseInstanceStudentAnalytics.js:17]: auth protect this route

**Review by JParisFerrer — CHANGES_REQUESTED** [review]

File: src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js (hunk 3 above)
Comment (Binary-Eater) [reply src/pages/courseInstanceAnalytics/courseInstanceAnalytics.js:17]: I assume we will be using middleware to protect all routes in the end @nwalters512 ?

**Review by JParisFerrer — APPROVED** [review]

**Pull request merged**